src/fosslight_dependency/package_manager/Yarn.py

```
@@ -42,13 +43,30 @@ def detect_yarn_version(self):
         return None
 
+    def install_yarn(self):
+        logger.info("Install yarn")
+        NODE_DIR = os.path.join(self.input_dir, ".fosslight_node")
+        NODE_BIN = NODE_DIR if self.IS_WINDOWS else os.path.join(NODE_DIR, "bin")
+        os.makedirs(NODE_DIR, exist_ok=True)
+        PATH_SEP = ";" if self.IS_WINDOWS else ":"
+        os.environ["PATH"] = NODE_BIN + PATH_SEP + os.environ.get("PATH", "")
+        yarn_install_cmd = f"npm install -g yarn --prefix {NODE_DIR}"
+
+        result = subprocess.run(yarn_install_cmd, shell=True, capture_output=True, text=True)
```

Ruff (0.15.7)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)